processingchain/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/processingchain/ProcessingChainBDTopoTest.groovy

```diff
@@ -58,10 +52,39 @@ class ProcessingChainBDTopoTest extends ChainProcessAbstractTest{
         }
     }
 
+    def loadFiles(String inseeCode){
+        H2GIS h2GISDatabase = H2GIS.open(h2db, "sa", "")
+        String relativePath = bdtopoFoldName + File.separator + inseeCode
+
+        // Test whether there is a folder containing .shp files for the corresponding INSEE code
+        if(ProcessingChain.class.getResource(relativePath) != null){
+            // Test is the URL is a folder
+            if(new File(ProcessingChain.class.getResource(relativePath).toURI()).isDirectory()){
+                listTables.each {
+                    String filePath = ProcessingChain.class.getResource(relativePath + File.separator + it + ".shp")
+                    // If some layers are missing, do not try to load them...
+                    if (filePath != null) {
+                        h2GISDatabase.load(filePath, it, true)
+                    }
+                }
+            }
+            else{
+                logger.error  "There is no folder containing shapefiles for commune insee $inseeCode"
```

Comment:
If in this case, the database is not well generated and doesn't contains the needed data, you can do a `fail()` (from `org.junit.jupiter.api`) to indicate to JUnit that there is an error and stop the test here.

Reply:
along with the logger or only use the fail ?

Reply:
Along with the logger.